src/EditorFeatures/CSharpTest/Formatting/CodeCleanupTests.cs

```diff
@@ -252,6 +246,41 @@ private void Method()
             return AssertCodeCleanupResult(expected, code);
         }
 
+        [Fact]
+        [Trait(Traits.Feature, Traits.Features.CodeCleanup)]
+        public Task FixMisplacedUsingDirectives()
```

Comment:
can we have another test in the oppsite direction.

Reply:
what do you mean, outside to inside?

Reply:
What do you mean, outside to inside?
I'll do that but it'll require expanding the base test method with another option for that.
Plus adding another option to [*GenerationOptions.cs*](https://github.com/dotnet/roslyn/blob/e16d3906a6d6a560da72144a942dbac7c8dc2e11/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Editing/GenerationOptions.cs).

Reply:
Almost all of the tests in that class are one-directional, except for these two: `systemUsingsFirst`, and `separateUsingGroups`.

The problem is that `AddImportPlacement` is internal, and can't be used as a parameter on this test method.

https://github.com/dotnet/roslyn/blob/91c6b154373d0835c69c889ecd27fa6e6cf29dc2/src/EditorFeatures/CSharpTest/Formatting/CodeCleanupTests.cs#L255

Reply:
you could always use a boolean on the method level and then translate that to the internal enum value inside the core method.

Reply:
There are 8 tests in that file, where only these two are defined in *GenerationOptions.cs*. That file has 55 references. Do you really want me to change that file?

Reply:
Even changing the default value inline to `Inside` doesn't do the inverse.

https://github.com/dotnet/roslyn/blob/e16d3906a6d6a560da72144a942dbac7c8dc2e11/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/CodeStyle/CSharpCodeStyleOptions.cs#L265-L276

Reply:
OK after playing with it for a while, seems that the `Inside` option, doesn't actively move namespaces inside. It's just that it doesn't move out the nested namespaces.

Updated test accordingly.

Reply:
> and can't be used as a parameter on this test method.

In cases like this, one option is to make the method `private protected` instead.

Reply:
> ... seems that the Inside option, doesn't actively move namespaces inside ...

The test `WhenInsidePreferred_UsingsInCompilationUnit_UsingsMovedAndSystemPlacedFirstIgnored` suggests that the setting is capable of moving the using directives. We should figure out why it is not applying here.

/cc @JoeRobich